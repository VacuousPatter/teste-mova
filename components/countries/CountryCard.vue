<template>
  <v-hover v-slot="{hover}">
    <v-card class="countries__country" link nuxt :to="`/country/${country.cca3||country.alpha3Code}`">
      <v-responsive :aspect-ratio="16/9">
        <v-img :aspect-ratio="16/9" :src="country.flags.png || country.coatOfArms.png" />
      </v-responsive>
      <v-expand-transition>
        <div
          v-if="hover"
          class="country__hover-panel"
        >
          <p>{{ country.name.common }} | {{ country.cca3 }}</p>
        </div>
      </v-expand-transition>
    </v-card>
  </v-hover>
</template>

<script>
export default {
  props: {
    country: {
      type: Object,
      required: true
    }
  }
}
</script>

<style scoped>
.country__hover-panel{
  position: absolute;
  bottom: 0px;
  left: 0px;
  width: 100%;
  background: #fff;
}

.country__hover-panel p{
  margin: 15px;
  font-size: 1.1em;
}
</style>
