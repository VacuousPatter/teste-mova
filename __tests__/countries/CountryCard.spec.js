import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils'
import Vuetify from 'vuetify'
import CountryCard from '@/components/countries/CountryCard.vue'
const data = require('./CountryCard.data.json')

describe('/components/countries/CountryCard.vue', () => {
  const paisTeste = data.pais

  describe('caso 1', () => {
    const localVue = createLocalVue()
    const vuetify = new Vuetify()
    const wrapper = mount(CountryCard, {
      localVue,
      vuetify,
      propsData: {
        country: paisTeste
      },
      stubs: {
        NuxtLink: RouterLinkStub
      },
      data: () => ({
        hovering: true
      })
    })

    test('hover mostra nome do pais', () => {
      const p = wrapper.findAll('p.pais--nome')
      expect(p.exists()).toBeTruthy()
    })

    test('snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('caso 2', () => {
    const localVue = createLocalVue()
    const vuetify = new Vuetify()
    const wrapper = mount(CountryCard, {
      localVue,
      vuetify,
      propsData: {
        country: paisTeste
      },
      stubs: {
        NuxtLink: RouterLinkStub
      },
      data: () => ({
        hovering: false
      })
    })

    test('sem hover nÃ£o mostra nome do pais', () => {
      const p = wrapper.findAll('p.pais--nome')
      expect(p.exists()).toBeFalsy()
    })

    test('snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
