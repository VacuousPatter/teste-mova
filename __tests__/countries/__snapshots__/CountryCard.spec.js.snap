// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/countries/CountryCard.vue caso 1 snapshot 1`] = `
"<a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
  <div class=\\"v-responsive\\">
    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
    <div class=\\"v-responsive__content\\">
      <div class=\\"v-image v-responsive theme--light\\">
        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
        </transition-stub>
        <div class=\\"v-responsive__content\\"></div>
      </div>
    </div>
  </div>
  <transition-stub name=\\"expand-transition\\">
    <div class=\\"country__hover-panel\\">
      <p class=\\"pais--nome\\">
        Aruba | ABW
      </p>
    </div>
  </transition-stub>
</a>"
`;

exports[`/components/countries/CountryCard.vue caso 2 snapshot 1`] = `
"<a tabindex=\\"0\\" class=\\"countries__country v-card v-card--link v-sheet theme--light\\">
  <div class=\\"v-responsive\\">
    <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
    <div class=\\"v-responsive__content\\">
      <div class=\\"v-image v-responsive theme--light\\">
        <div class=\\"v-responsive__sizer\\" style=\\"padding-bottom: 56.25%;\\"></div>
        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
        </transition-stub>
        <div class=\\"v-responsive__content\\"></div>
      </div>
    </div>
  </div>
  <transition-stub name=\\"expand-transition\\">
    <!---->
  </transition-stub>
</a>"
`;
